### ISSUE TITLE
edmf random seed not too random?

### ISSUE THREAD

**Issue by mathomp4** [GEOS-ESM/GEOSgcm_GridComp#614]
@narnold1 A question. While talking with NVIDIA about their GPU efforts with edmf we looked at this code:

https://github.com/GEOS-ESM/GEOSgcm_GridComp/blob/b3b07baef35ffe29211d34c642b77aaef1a0a2a0/GEOSagcm_GridComp/GEOSphysics_GridComp/GEOSturbulence_GridComp/edmf.F90#L347-L356

and then `the_seed` is passed to `Poisson`:

https://github.com/GEOS-ESM/GEOSgcm_GridComp/blob/b3b07baef35ffe29211d34c642b77aaef1a0a2a0/GEOSagcm_GridComp/GEOSphysics_GridComp/GEOSturbulence_GridComp/edmf.F90#L362

The thing is...I'm pretty sure `the_seed(1)` equals `the_seed(2)` unless they are 0. We were thinking maybe it should be:
```fortran
 THE_SEED(1)=seedmf(1) + seedmf(2) 
 THE_SEED(2)=seedmf(1) - seedmf(2) 
```
at the beginning?

Just wondering.

Cc: @tclune as he probably knows more about algorithmically making random numbers than I do

**Comment by tclune** [comment]
It all looks very ad hoc to me.   Some attempt to pre-randomize a seed?

**Comment by narnold1** [comment]
This section of code was copied from ras.F90.  I'm not sure what motivated lines 351-356.  Is THE_SEED=0 disallowed?  I think the current code works, in the sense of producing randomness and satisfying regression tests, but I will double check that.

**Comment by mathomp4** [comment]
> This section of code was copied from ras.F90. I'm not sure what motivated lines 351-356. Is THE_SEED=0 disallowed? I think the current code works, in the sense of producing randomness and satisfying regression tests, but I will double check that.

@narnold1 There are compilers where setting a seed that's not too random causes issues (see https://forums.developer.nvidia.com/t/random-number-seed-oddities/133398 by some "Matt Thompson"), so I always discourage that once I realized that. I mean, setting the seed based on input temperatures isn't bad, it's just like the seed construction was a bit oddly coded. 😄

**Comment by narnold1** [comment]
This issue is addressed by PR #783
